nuclio/triggers.py

```diff
@@ -225,24 +225,66 @@ class V3IOStreamTrigger(NuclioTrigger):
     def __init__(
         self,
         url: str = None,
-        seekTo: str = "latest",
+        seek_to: str = "latest",
         partitions: list = None,
-        pollingIntervalMS: int = 500,
-        readBatchSize: int = 64,
-        maxWorkers: int = 1,
+        polling_interval_ms: int = 500,
+        read_batch_size: int = 64,
+        max_workers: int = 1,
         access_key: str = None,
-        sessionTimeout: str = "10s",
+        session_timeout: str = "10s",
         name: str = None,
         container: str = None,
         path: str = None,
-        workerAllocationMode: str = "pool",
+        worker_allocation_mode: str = "pool",
         webapi: str = Constants.default_webapi_address,
-        consumerGroup: str = "default",
-        sequenceNumCommitInterval: str = "1s",
-        heartbeatInterval: str = "3s",
+        consumer_group: str = "default",
+        sequence_num_commit_interval: str = "1s",
+        heartbeat_interval: str = "3s",
         explicit_ack_mode: str = None,
         extra_attributes=None,
+        **deprecated_kwargs,
     ):
+        # TODO: delete deprecated arguments
+        deprecation_warning_template = "Deprecated argument '{old_arg_name}' will be removed in a future version. " \
+                                       "Please use '{new_arg_name}' instead."
+
+        if "seekTo" in deprecated_kwargs:
+            logger.warn(deprecation_warning_template.format(old_arg_name="seekTo",
+                                                            new_arg_name="seek_to"))
+            seek_to = deprecated_kwargs.get("seekTo")
```

Comment:
@gtopper yes, that's true, but I'm not sure if we need to overcomplicate it. @TomerShor  wdyt?

Reply:
I would prefer to prioritize the new snake_case arguments

Reply:
It was decided not to proceed with prioritisation of snake_case args, because we have default values for each argument, therefore we can't easily check here for `None`.

Reply:
I think that's ok, but I'll point out that we could remove the default arguments and only set the defaults in the method body (this is the style used in mlrun, I suppose exactly because this type of situation).